docs/docs/unreachable.md

```diff
@@ -0,0 +1,143 @@
+
+### Agent Unreachable Rate Limiter
+
+In the previous section, we mentioned that Mesos will consider an agent inactive, assuming that only 1 agent is lost. This is because Apache Mesos has a agent rate limiter which was established prior to partition aware frameworks and is still enforced. The purpose of the rate limiter is to reduce the number of reported lost agents within a specified amount of time through the `--agent_removal_rate_limit` flag.
+
+In Mesos, the default for `--agent_removal_rate_limit`is `None`, which has the effect of reporting agents immediately after the agent health check logic as described above.
+
+The default in DC/OS however is [1/20mins](https://github.com/dcos/dcos/blob/9cc6ab28060545cd203c09aa7fa1b9456773d080/gen/dcos-config.yaml#L442)
+which is 1 agent every 20 minutes. The following hypothetical timeline describes how 2 agents that become lost at the same time will be marked inactive:
+
+- 12:00:00 - Agent 1 goes off-line
+- 12:03:00 - Agent 2 goes off-line
+- 12:20:00 - Mesos Master marks agent 1 as inactive, and publishes a `TASK_UNREACHABLE` status update for relevant tasks.
+- 12:40:00 - Mesos Master marks agent 2 as inactive, and publishes a `TASK_UNREACHABLE` status update for relevant tasks.
+
+The net result is that the amount of time that a agent on DC/OS can be lost or unreachable before Marathon is notified is not always deterministic. It is defined by the number of lost nodes, and the order in which they were lost [^1]. It is important to understand that these tasks are listed by Mesos as Active until this time and Marathon is completely unaware until notified by Mesos.
+
+**NOTE:**
+There are 2 JIRAs against Mesos to remove rate limits. One marked as a critical bug ([MESOS-7721](https://issues.apache.org/jira/browse/MESOS-7721))
+the other as a major improvement ([MESOS-5948](https://issues.apache.org/jira/browse/MESOS-5948)). There is currently discussion in process to remove this rate limit default from DC/OS.
+
+## Marathon Unreachable Strategy
+
+Marathon has configuration options for working with unreachable tasks by setting the unreachable strategy for a Marathon application as part as the app definition:
+
+```json
+unreachableStrategy: {
+  "inactiveAfterSeconds": 0
+}
+```
+
+In order for this to take effect it is necessary to get a `TASK_UNREACHABLE` status update from Apache Mesos. The above constraints set the minimum time of managing an unreachable strategy event. The configuration for the app definition sets up to time events. The first is `inactiveAfterSeconds` which is the time in seconds after the `TASK_UNREACHABLE` event from Mesos that Marathon will create a replacement task. The second configuration is `expungeAfterSeconds` which is the amount of time after the `TASK_UNREACHABLE` that a task will be removed killed provided:
+
+1. it was replaced with the `inactiveAfterSeconds` event
+2. The original task (which was unreachable) becomes reachable again.
```

Comment:
Hm. This is a little misleading. It is only killed if it comes back after `expungeAfterSeconds` passed.